libwayshot/src/lib.rs

```diff
@@ -491,15 +493,23 @@ impl WayshotConnection {
 
                 debug!("Committing surface with attached buffer.");
                 surface.commit();
-
+                layer_shell_surfaces.push((surface, layer_surface));
                 event_queue.blocking_dispatch(&mut state)?;
 
                 Ok(())
             })?;
         }
+
         let callback_result = callback();
-        layer_shell.destroy();
-        event_queue.blocking_dispatch(&mut state)?;
+
+        debug!("Unmapping and destroying layer shell surfaces.");
+        for (surface, layer_shell_surface) in layer_shell_surfaces.iter() {
+            surface.attach(None, 0, 0);
+            surface.commit(); //unmap surface by committing a null buffer
```

Comment:
Note: I am approving and merging the PR for now but I am unsure if this is correct